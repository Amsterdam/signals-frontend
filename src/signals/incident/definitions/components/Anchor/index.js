import React from 'react';
import PropTypes from 'prop-types';

const Anchor = ({
  href, target, className, children,
}) => <a href={href} className={className} target={target}>{children}</a>;

Anchor.defaultProps = {
  target: '_self',
  className: '',
};

Anchor.propTypes = {
  children: PropTypes.any,
  href: PropTypes.string.isRequired,
  target: PropTypes.string,
  className: PropTypes.string,
};

export default Anchor;
