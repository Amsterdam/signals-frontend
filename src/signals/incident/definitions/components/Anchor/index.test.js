import React from 'react';
import { shallow } from 'enzyme';

import Anchor from './index';

describe('Definition component <Anchor />', () => {
  let wrapper;

  describe('rendering', () => {
    it('should render with text and text correctly', () => {
      wrapper = shallow(<Anchor href="http://amsterdam.nl">Dit is een link naar amsterdam</Anchor>);

      expect(wrapper).toMatchSnapshot();
    });

    it('should render with text and text and target correctly', () => {
      wrapper = shallow(<Anchor href="http://amsterdam.nl" target="_blank">Dit is een link naar amsterdam</Anchor>);

      expect(wrapper).toMatchSnapshot();
    });
  });
});
