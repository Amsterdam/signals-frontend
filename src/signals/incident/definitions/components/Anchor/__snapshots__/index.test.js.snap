// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Definition component <Anchor /> rendering should render with text and text and target correctly 1`] = `
<a
  className=""
  href="http://amsterdam.nl"
  target="_blank"
>
  Dit is een link naar amsterdam
</a>
`;

exports[`Definition component <Anchor /> rendering should render with text and text correctly 1`] = `
<a
  className=""
  href="http://amsterdam.nl"
  target="_self"
>
  Dit is een link naar amsterdam
</a>
`;
